### DISCUSSION TITLE
[autolink] Reference commit SHA in a commit title via PAT

### DISCUSSION THREAD

**Discussion by MMirelli** [community/community#137602] [Programming Help] [answered]
### Select Topic Area

Question

#### Context
I am trying to get **autolinked references** to commits to work inside of a commit title, when they are generated by a bot using a fine-grained PAT. The end goal is to have a bot that would commit some changes in a private repository (let's call it `r1`) and generate a commit title containing an autolinked reference to the SHA of a commit in another private repository (`r2`). Both repositories live in the same GH organization (let's say `o`). So to recap the bot produces a commit in `o/r1`, referencing a commit SHA (`shabc`) in repository `o/r2` using the syntax `o/r2@shabc`, as per [the 4th row in the table](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls#commit-shas).

##### Details
The bot uses a fine-grained PAT to push the changes to the private repository. Also, I have ensured the PAT has at least read access to commits, setting the following PAT permissions to read-only for both `o/r1` and `o/r2`:
1. `Commit statuses`
2. `Contents`

This is because, [as per this GH doc](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls#troubleshooting-referencing-commit-shas), GH demands at least read access to the referenced commit (in this case only `o/r2`).

#### Expectation
`o/r2@shabc` is shortlinked to the actual commit in `o/r2`, which is not happening.

Any ideas to further troubleshoot the issue would be very helpful.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @MMirelli ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by MMirelli** [comment, marked as answer]
This works actually. The issue was caused by a git misconfiguration: the bot was not using the email setup for its account, but some other email.
